sensing/src/main/java/com/google/android/sensing/SensingEngineProvider.kt

```diff
@@ -55,19 +56,27 @@ object SensingEngineProvider {
   fun getBlobStoreService(): BlobstoreService {
     if (blobstoreService == null) {
       with(sensingEngineConfiguration!!.serverConfiguration) {
-        blobstoreService =
-          BlobstoreService(
-            MinioAsyncClient.builder()
-              .endpoint(baseUrl)
-              .credentials(authenticator!!.getUserName(), authenticator.getPassword())
-              .httpClient(
-                OkHttpClient.Builder()
-                  .connectTimeout(networkConfiguration.connectionTimeOut, TimeUnit.SECONDS)
-                  .writeTimeout(networkConfiguration.writeTimeOut, TimeUnit.SECONDS)
-                  .build()
-              )
-              .build()
+        val okHttpClient =
+          OkHttpClient.Builder()
+            .connectTimeout(networkConfiguration.connectionTimeOut, TimeUnit.SECONDS)
+            .writeTimeout(networkConfiguration.writeTimeOut, TimeUnit.SECONDS)
+            .build()
+        val minioClientBuilder = MinioAsyncClient.builder().endpoint(baseUrl)
```

Comment:
Other way of implementing this could be, expose a `ProviderFactory` class which initializes the `Provider` and returns it. For `BasicAuthenticator` you could also use `MinioEnvironmentProvider` as well. See if it can be more of config driven, so that the clients does not have to write much code.

Reply:
Even with config, application will need to write code to provide with token, roleArn, durationsSeconds, etc. 
Instead of sensing library picking things from config + code, let's stick to code only.